content/docs/configuration/acme/dns01/route53.md

```diff
@@ -47,30 +46,392 @@ permissions:
+
+   To use IRSA with cert-manager you must first enable the feature for your cluster.
+   Follow the [official documentation](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html).
+
+2. **Create a trust relationship**
+
+   In this configuration an IAM role is mapped to the cert-manager `ServiceAccount` allowing it to authenticate with AWS.
+   The IAM role you map to the `ServiceAccount` will need permissions on any and all Route53 zones cert-manager will be using.
+   Create a trust relationship by adding the following trust policy to the IAM role:
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Action": "sts:AssumeRoleWithWebIdentity",
+         "Principal": {
+           "Federated": "arn:aws:iam::<aws-account-id>:oidc-provider/oidc.eks.<aws-region>.amazonaws.com/id/<eks-hash>"
+         },
+         "Condition": {
+           "StringEquals": {
+             "oidc.eks.<aws-region>.amazonaws.com/id/<eks-hash>:sub": "system:serviceaccount:<namespace>:<service-account-name>"
+           }
+         }
+       }
+     ]
+   }
+   ```
+
+   Replace the following:
+
+   - `<aws-account-id>` with the AWS account ID of the EKS cluster.
+   - `<aws-region>` with the region where the EKS cluster is located.
+   - `<eks-hash>` with the hash in the EKS API URL; this will be a random 32 character hex string (example: `45DABD88EEE3A227AF0FA468BE4EF0B5`).
+   - `<namespace>` with the namespace where cert-manager is running.
+   - `<service-account-name>` with the name of the `ServiceAccount` object created by cert-manager.
+
+
+   > ℹ️ If you're following the Cross Account example, this trust policy is attached to the cert-manager role in Account X with ARN `arn:aws:iam::XXXXXXXXXXX:role/cert-manager`.
+   > The permissions policy is the same as above.
+
+3. **Annotate the cert-manager `ServiceAccount`**
+
+   ```yaml
+   apiVersion: v1
+   kind: ServiceAccount
+   metadata:
+     annotations:
+       eks.amazonaws.com/role-arn: arn:aws:iam::XXXXXXXXXXX:role/cert-manager
+   ```
+
+   The cert-manager Helm chart provides a variable for injecting annotations into cert-manager's `ServiceAccount` like so:
+
+   ```yaml
+   serviceAccount:
+     annotations:
+       eks.amazonaws.com/role-arn: arn:aws:iam::XXXXXXXXXXX:role/cert-manager
+   ```
+
+   > ℹ️ If you're following the Cross Account example, modify the `ClusterIssuer` with the role from Account Y.
+
+3. **(optional) Update file system permissions**
+
+   You may also need to modify the cert-manager `Deployment` with the correct file system permissions, so the `ServiceAccount` token can be read.
+
+   ```yaml
+   spec:
+     template:
+       spec:
+         securityContext:
+           fsGroup: 1001
+   ```
+
+   The cert-manager Helm chart provides a variable for modifying cert-manager's `Deployment` like so:
+
+   ```yaml
+   securityContext:
+     fsGroup: 1001
+   ```
```

Comment:
I have labelled this as optional, because I did not need  this when I  tested IRSA on an EKS cluster.
The reason may be found in https://github.com/cert-manager/cert-manager/issues/2147#issuecomment-780212025 where @TBBle wrote:
> Mentioned on the [k8s 1.19 for EKS release notes](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-1.19), I believe the fsGroup setting should not be needed for k8s 1.19 onwards, due to the implementation of https://github.com/kubernetes/enhancements/pull/1598.

I think it may be specifically required when using Fargate, based on @iainlane 's feedback in https://github.com/cert-manager/website/issues/697.
I will leave it to Fargate users to help us clarify this part of the documentation in future PRs.

Reply:
Added this note to encourage users to tell us once and for all whether this step is obsolete.

@SgtCoDFish I see that you also asked whether this step is necessary:
 *  https://github.com/cert-manager/website/pull/698#issuecomment-915304069